test/e2e/util_test.go

```diff
@@ -73,25 +74,25 @@ func newNamespaceCleaner(namespace string) *namespaceCleaner {
 }
 
 // notifyOnFailure checks if a test has failed or cleanup is true before cleaning a namespace
-func (c *namespaceCleaner) NotifyTestComplete(t GinkgoTInterface, cleanup bool) {
-	if t.Failed() {
+func (c *namespaceCleaner) NotifyTestComplete(t GinkgoTestDescription, cleanup bool) {
+	if t.Failed {
 		c.skipCleanupOLM = true
 	}
 
 	if c.skipCleanupOLM || !cleanup {
-		t.Log("skipping cleanup")
+		ctx.Ctx().Logf("skipping cleanup")
 		return
 	}
 
-	cleanupOLM(t, c.namespace)
+	cleanupOLM(c.namespace)
 }
 
 // newKubeClient configures a client to talk to the cluster defined by KUBECONFIG
-func newKubeClient(t GinkgoTInterface) operatorclient.ClientInterface {
+func newKubeClient() operatorclient.ClientInterface {
```

Comment:
I was thinking of removing this wrapper method that returns a client and directly get the client from the `ctx` object? Thoughts?

Reply:
Let's leave the `new*Client` functions in place for now. I don't think the `ctx` API has settled yet, so increasing the number of places that depend on it may end up making it harder to change (it's completely fine to be using `ctx.Ctx()` as you have done elsewhere in this PR).